services/movie_session.py

```diff
@@ -0,0 +1,53 @@
+import init_django_orm  # noqa: F401
+
+from datetime import datetime
+
+from django.db.models import QuerySet
+
+from db.models import MovieSession
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id
+    )
+
+
+def get_movies_sessions(session_date: datetime = None) -> QuerySet:
+    if session_date:
+        return MovieSession.objects.filter(show_time__date=session_date)
+
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> None:
+
+    if show_time:
+        MovieSession.objects.filter(id=session_id).update(show_time=show_time)
+
+    if movie_id:
+        MovieSession.objects.filter(id=session_id).update(movie_id=movie_id)
+
+    if cinema_hall_id:
+        MovieSession.objects.filter(id=session_id).update(
+            cinema_hall_id=cinema_hall_id
+        )
```

Comment:
The update logic for `update_movie_session` can be optimized by using a single `update` call with a dictionary of fields to update. This reduces the number of database queries and improves performance.